### PR TITLE
feat: implement -n X and -a flags

### PR DESCRIPTION
**Pull request by tedim52**
## Description:
This PR adds the `-a` and `-n` flags to `kurtosis service logs`.

- `-a` flag returns all logs of a service
- `-n X` returns `X` number of log lines

`-a` used with `-n X` will return all the logs. Both flags can be used in combination with `-f`. `... service logs` without any flags by default returns 200 log lines.

## Is this change user facing?
YES

### CODE REVIEW

File: engine/server/engine/centralized_logs/client_implementations/persistent_volume/stream_logs_strategy/per_week_stream_logs_strategy.go [hunk 1]
```diff
@@ -198,39 +146,141 @@ func (strategy *PerWeekStreamLogsStrategy) getRetainedLogsFilePaths(filesystem v
...
+		select {
+		case <-ctx.Done():
+			logrus.Debugf("Context was canceled, stopping streaming service logs for service '%v'", serviceUuid)
+			return nil
+		default:
+			jsonLogStr, err := getCompleteJsonLogString(logsReader)
+			if isValidJsonEnding(jsonLogStr) {
+				// collect all log lines in tail log lines
+				tailLogLines = append(tailLogLines, jsonLogStr)
+				if len(tailLogLines) > int(numLogLines) {
```
Comment (Peeeekay) [inline engine/server/engine/centralized_logs/client_implementations/persistent_volume/stream_logs_strategy/per_week_stream_logs_strategy.go:226]: I guess you are handling the empty string issue, in next PR. Make sense.

**Comment by Peeeekay** [comment]
@tedim52 Looks good to me. I am trusting the tests

**Review by Peeeekay — APPROVED** [review]

File: engine/server/engine/centralized_logs/client_implementations/persistent_volume/stream_logs_strategy/per_week_stream_logs_strategy.go (hunk 1 above)
Comment (tedim52) [reply engine/server/engine/centralized_logs/client_implementations/persistent_volume/stream_logs_strategy/per_week_stream_logs_strategy.go:226]: it's not too explicit but actually, the `isValidJsonEnding` check should ensure that no empty log lines are streamed!

**Comment by leeederek** [comment]
@tedim52 - i cannot see the docs for this, do you mind adding?

**Pull request merged**